packages/@aws-cdk/cdk/lib/fn.ts

```diff
@@ -292,14 +293,14 @@ export class Fn {
    * @returns a token represented as a string array
    */
   public valueOfAll(parameterType: string, attribute: string): string[] {
-    return new FnValueOfAll(parameterType, attribute).toList();
+    return Token.encodeAsList(new FnValueOfAll(parameterType, attribute));
```

Comment:
At this point I don't think we need all the FnXxx private classes any more. Let's just embed the intrinsic function directly.

Reply:
Maybe, but they're all private so not really hurting.